modules/raft/src/main/java/org/apache/ignite/raft/jraft/NodeManager.java

```diff
@@ -134,22 +136,31 @@ private void onSentHeartbeat() {
                     queue.addAll(blocked);
 
                     try {
-                        rpcClient.invokeAsync(peer, builder.build(), null, (result, err) -> {
-                            if (err != null) {
-                                for (CompletableFuture<Message> fut : futs) {
-                                    fut.completeExceptionally(err);
+                        rpcClient.invokeAsync(peer, builder.build(), null, new InvokeCallback() {
+                            @Override
+                            public void complete(Object result, Throwable err) {
+                                if (err != null) {
+                                    for (CompletableFuture<Message> fut : futs) {
+                                        fut.completeExceptionally(err);
+                                    }
+
+                                    return;
                                 }
 
-                                return;
-                            }
+                                CoalescedHeartbeatResponse resp = (CoalescedHeartbeatResponse) result;
+
+                                assert resp.messages().size() == futs.size();
 
-                            CoalescedHeartbeatResponse resp = (CoalescedHeartbeatResponse) result;
+                                int i = 0;
+                                for (Message message : resp.messages()) {
+                                    futs.get(i++).complete(message); // Future completion will trigger callbacks.
+                                }
 
-                            assert resp.messages().size() == futs.size();
+                            }
 
-                            int i = 0;
-                            for (Message message : resp.messages()) {
-                                futs.get(i++).complete(message); // Future completion will trigger callbacks.
+                            @Override
+                            public Executor executor() {
```

Comment:
Let's change this method to be non default,  so called always has to provide the executor.